.github/workflows/mpi_matrix_test.yml

```diff
@@ -9,29 +9,67 @@ on:
       - master
 
 jobs:
-  build:
+  pyomo-mpi-tests:
     name: ${{ matrix.TARGET }}/py${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
     strategy:
-      max-parallel: 1
+      fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.7]
         include:
         - os: ubuntu-latest
           TARGET: linux
+        python-version: [3.7]
 
     steps:
     - uses: actions/checkout@v2
+
+    - name: OS package cache
+      id: pkg-cache
+      uses: actions/cache@v1
+      with:
+        path: pkg-cache
+        key: pkg-v1-${{runner.os}}
+
+    - name: Download cache
+      id: download-cache
+      uses: actions/cache@v1
+      with:
+        path: download-cache
+        key: download-v1-${{runner.os}}
+
     - name: Setup conda environment
       uses: s-weigand/setup-conda@v1
       with:
-        update-conda: true
         python-version: ${{ matrix.python-version }}
+        update-conda: true
         conda-channels: anaconda, conda-forge
 
     - name: Install dependencies
       run: |
+        # Ensure cache directories exist
+        mkdir -p ${GITHUB_WORKSPACE}/download-cache
+        mkdir -p ${GITHUB_WORKSPACE}/pkg-cache
+        if test "${{matrix.TARGET}}" == osx; then
```

Comment:
Unless there are plans to run MPI tests on OSX, this can be deleted. (No need for cruft!)

Reply:
I was torn about leaving that in, too.  In the end, I elected to leave it in to minimize the diffs between `mpi_matrix_test.yml` and `unix_python_matrix_test.yml` (which should make keeping them in sync with each other easier).

Reply:
Conversely, it causes a bit more work if there is a change made on the OSX side but not the Linux side. Because then we have to update a bit of code that isn't being used anyway, or run the risk of the outdated version lingering in the MPI file.

Reply:
Yeah.  What do you think about maintaining patch files in the repo, so that we (in theory) just edit `unix_python_matrix_test.yml` and then apply the patch files to generate `mpi_matrix_test.yml` and `push_branch_unix_test.yml`?  It has bothered me that we have (basically) the same code in three locations...

Reply:
It also irks me. I like that idea! We need to make sure to document it for the Developers.

Reply:
@jsiirola - Gorgeous update on this. I like it a lot.